addons/l10n_id/models/qris_transaction.py

```diff
@@ -0,0 +1,98 @@
+
+class QrisTransaction(models.Model):
+    _name = "l10n_id.qris.transaction"
+    _description = "Records of QRIS Transactions"
+
+    model = fields.Char(readonly=True)
+    res_id = fields.Integer(readonly=True)
+    partner_bank_id = fields.Many2one('res.partner.bank', string='Recipient Bank')
+
+    # QRIS related information
+    qris_content = fields.Char("QR Content")
+    qris_invoiceid = fields.Char("QRIS Transaction Number")
+    trx_date = fields.Datetime("Transaction Creation Time")
+    paid = fields.Boolean()
+    amount = fields.Float()
+
+    _sql_constraints = [('ref_uniq', "unique(model, res_id)",
+                         "There's already a transaction with the same reference")]
+
+    def _create_payment_vals(self, move, **extra_create_values):
+        # reference to _create_payment from payment.transaction
+        return {
+            'amount': self.amount,
+            'payment_type': 'inbound',
+            'currency_id': move.currency_id.id,
+            'partner_id': move.partner_id.commercial_partner_id.id,
+            'partner_type': 'customer',
+            'journal_id': self.partner_bank_id.journal_id.id,
+            'ref': move.payment_reference,
+            **extra_create_values
+        }
+
+    def _cron_fetch_payment_status(self):
+        """ Get all qris transaction that is within 30 minutes and fetch each status"""
+        prev_30_minutes = fields.Datetime.now() - datetime.timedelta(minutes=30)
+        transactions = self.search([('paid', '=', False), ('trx_date', '>=', prev_30_minutes), ('model', '=', 'account.move')])
+        paid_transactions = transactions.fetch_payment_status()
+        paid_transactions.register_payment()
+        self._clean_transactions()
+
+    def fetch_payment_status(self):
+        """ Fetch the status of each transaction via API
+        """
+        paid_transactions = self.env['l10n_id.qris.transaction']
+        for transaction in self:
+            params = {
+                'do': 'checkStatus',
+                'apikey': transaction.partner_bank_id.l10n_id_qris_api_key,
+                'mID': transaction.partner_bank_id.l10n_id_qris_mid,
+                'invid': transaction.qris_invoiceid,
+                'trxvalue': int(transaction.amount),
+                'trxdate': transaction.trx_date.strftime('%Y-%m-%d')
+            }
+            response = self.partner_bank_id.make_qris_request(QRIS_FETCH_STATUS_URL, params)
+
+            # upate payment status of status repsonse is 'paid'
+            _logger.debug("Found response: %s", pprint.pformat(response))
+            payment_status = response['data'].get('qris_status', 'unpaid')
+
+            if payment_status == "paid":
+                transaction.paid = True
+                paid_transactions |= transaction
+            elif payment_status == "unpaid":
+                transaction.paid = False
+
+        return paid_transactions
+
+    def register_payment(self):
```

Comment:
```suggestion
    def _register_payment(self):
```
I Don't think there is any need for this to be public? Same with the other methods.